cluster/sync.sh

```diff
@@ -2,80 +2,106 @@
 
 set -ex
 
+kubectl=./cluster/kubectl.sh
+
 function getDesiredNumberScheduled {
-    echo $(./cluster/kubectl.sh get daemonset -n nmstate $1 -o=jsonpath='{.status.desiredNumberScheduled}')
+    $kubectl get daemonset -n nmstate $1 -o=jsonpath='{.status.desiredNumberScheduled}'
 }
 
 function getNumberAvailable {
-    numberAvailable=$(./cluster/kubectl.sh get daemonset -n nmstate $1 -o=jsonpath='{.status.numberAvailable}')
+    numberAvailable=$($kubectl get daemonset -n nmstate $1 -o=jsonpath='{.status.numberAvailable}')
     echo ${numberAvailable:-0}
 }
 
+function eventually {
+    timeout=15
+    interval=5
+    cmd=$@
+    echo "Checking eventually $cmd"
+    while ! $cmd; do
+        sleep $interval
+        timeout=$(( $timeout - $interval ))
+        if [ $timeout -le 0 ]; then
+            return 1
+        fi
+    done
+}
+
 function consistently {
+    timeout=15
+    interval=5
     cmd=$@
-    retries=3
-    interval=1
-    cnt=1
-    while [[ $cnt -le $retries ]]; do
-        $cmd
+    echo "Checking consistently $cmd"
+    while $cmd; do
         sleep $interval
-        cnt=$(($cnt + 1))
+        timeout=$(( $timeout - $interval ))
+        if [ $timeout -le 0 ]; then
+            return 0
+        fi
     done
 }
 
 function isOk {
     desiredNumberScheduled=$(getDesiredNumberScheduled $1)
     numberAvailable=$(getNumberAvailable $1)
+    [ "$desiredNumberScheduled" == "$numberAvailable" ]
+}
+
+deploy() {
+    # Cleanup previous deployment, if there is any
+    make cluster-clean
 
-    if [ "$desiredNumberScheduled" == "$numberAvailable" ]; then
-        echo "$1 DS is ready"
-        return 0
+    # Fetch registry port that can be used to upload images to the local kubevirtci cluster
+    registry_port=$(./cluster/cli.sh ports registry | tr -d '\r')
+    if [[ "${KUBEVIRT_PROVIDER}" =~ ^(okd|ocp)-.*$ ]]; then \
+            registry=localhost:$(./cluster/cli.sh ports --container-name=cluster registry | tr -d '\r')
     else
-        return 1
+        registry=localhost:$(./cluster/cli.sh ports registry | tr -d '\r')
     fi
-}
 
-# Cleanup previous deployment, if there is any
-make cluster-clean
-
```

Comment:
I like the refactoring, but it is really difficult to review this PR since it contains multiple unrelated changes. Please move this refactoring to a separate PR or at least a dedicated commit. Let's do it properly

Reply:
ooook